### ISSUE TITLE
OpenMP parallelizzation doesn't work on Apple Silicon M1Pro, M1Max and M1Ultra

### ISSUE THREAD

**Issue by Francyrad** [llvm/llvm-project#62554]
Dear developers of OpenMP
I was working with a N-Body code called Rebound. Is it possible to run the code using the parallelisation provided with OpenMP. It works perfectly with Intel and M1 chips, but it doesn't work with the chips that I mentioned above. I can get just a x1 scaling running the OpenMP example of rebound

https://rebound.readthedocs.io/en/latest/c_examples/openmp/

Is there a possibility to solve the problem? Thank you for your availability

**Comment by Francyrad** [comment]
Hello, are tehre any news? The bug is still present

**Comment by shiltian** [comment]
@Francyrad Can you let me how you compile the program and how you got the library? Did you compile everything (LLVM, OpenMP) from scratch?

**Comment by Francyrad** [comment]
Yes, i compiled everything using homebrew, the program compile nicely and the rebound's example runs,
. OpenMP is perfectly installed, but i get a scaling factor of 1x. That is the same for all other simulations that uses OpenMP. 

The problem is only present for Apple silicon M1 Pro, MAX and Ultra, not M1 (which can get a scaling of 4x)

I have no idea for the M2 series

I think that the problem may be related to this:

https://github.com/dmlc/xgboost/issues/8320

Thank you for the answer, i hope the bug will be solved.

**Comment by Francyrad** [comment]
The solution may be simpler (or i'm just misunderstanding something). In a M1 (only M1) machine the scaling factor work up to 4x, which is more or less what i get in an INTEL machine (not a perfect scaling, but better than nothing). So why that should work on m1 and not in M1Pro?
